addon/components/frost-button.js

```diff
@@ -100,7 +122,15 @@ export default Ember.Component.extend({
    * ['small', 'medium', 'large', 'extra-large']
    * @type {String}
    */
-  size: 'medium',
+  size: '',
```

Comment:
When `design` is present, we don't want `priority` and `size` to be specified.  If they are present, we will ignore them and print the warning.  Since they have a value by default, we would always end up with the warning even if the application did not specify it.  So it is a bit cleaner without the default values.